WikiPreview.Fluent.Plugin/WikiPreviewSearchApp.cs

```diff
@@ -108,59 +111,84 @@ public ValueTask LoadSearchApplicationAsync()
-                            PreviewImage = bitmapImageResult,
-                            DisplayedName = displayedName,
-                            ResultName = resultName,
-                            SearchedText = searchedText,
-                            Score = score,
-                            PageId = pageId,
-                            AdditionalInformation = additionalInfo
-                        };
-                        await channel.Writer.WriteAsync(wikiPreviewSearchResult).ConfigureAwait(false);
-                    }, cancellationToken).ContinueWith(_ => channel.Writer.Complete());
+                        }, cancellationToken)
+                        .ContinueWith(_ => channel.Writer.Complete(), CancellationToken.None);
                 else
                     channel.Writer.Complete();
-            });
+            }, CancellationToken.None);
 
             await foreach (WikiPreviewSearchResult item in channel.Reader.ReadAllAsync(cancellationToken))
                 yield return item;
         }
 
-        public ValueTask<ISearchResult> GetSearchResultForId(string serializedSearchObjectId)
+        public async ValueTask<WikiPreviewSearchResult> GetSearchResultForId(object serializedSearchObjectId)
         {
-            return new();
+            string pageId = serializedSearchObjectId as string;
+            if (string.IsNullOrWhiteSpace(pageId))
+                return default;
+
+            string url = "https://en.wikipedia.org/w/api.php?action=query&prop=extracts|pageimages&pageids=" +
+                         pageId +
+                         "&explaintext&exintro&pilicense=any&pithumbsize=100&format=json";
+
+            using var httpClient = new HttpClient();
+            var wiki = await httpClient.GetFromJsonAsync<Wiki>(url, _serializerOptions);
+            if (wiki == null) return default;
+
+            Dictionary<string, PageView>.ValueCollection pages = wiki.Query.Pages.Values;
+            if (pages is {Count: 0}) return default;
+
+            PageView pageView = pages.First();
+            WikiPreviewSearchResult wikiTask = await GenerateSearchResult(pageView, pageView?.Title);
+            return wikiTask;
+        }
+
+        private static async ValueTask<WikiPreviewSearchResult> GenerateSearchResult(PageView value,
+            string searchedText)
+        {
+            string resultName = value.Extract;
+            string displayedName = value.Title;
+            double score = displayedName.SearchDistanceScore(searchedText);
+            string pageId = value.PageId.ToString();
+            string wikiUrl = WikiRootUrl + displayedName.Replace(' ','_');
+            BitmapImageResult bitmapImageResult;
+
+            string additionalInfo = "";
+            if (!string.IsNullOrWhiteSpace(resultName))
+            {
+                using var reader = new StringReader(resultName);
+                additionalInfo = reader.ReadLine() ?? resultName;
```

Comment:
Don't use the same text for `AdditionalInfo` and `ResultName`, it will look weird. If the `ReadLine` results in null or empty data, put null in `AdditionalInfo`

Reply:
Yeah keeping the same content is not looking good. Shall I remove the `AdditionalInfo` all together? It looks more clean that way. As of now, I can't think of any other info to show.

Reply:
Need to find a way to differentiate between results with very similar name, you don't have to use `AdditionalInfo` you can also append it to the `DisplayedName`

Reply:
I'm not sure if this is what you wanted but Wikipedia API (and also Plugin) already gives search results with pair of braces if there are similar names. The braces can contain some extra terms to differentiate it.